### PR TITLE
Patching Changelogs

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

**Comment by weshaggard** [comment]
https://github.com/Azure/azure-sdk-for-go/blob/main/eng/scripts/Language-Settings.ps1#L4 that can now be removed as a result of this update.

**Review by chidozieononiwu — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Comment by weshaggard** [comment]
@ArcturusZhang is there anything in the arm automation that needs to be updated so that the "v" doesn't show up in the changelog entries?

**Comment by tadelesh** [comment]
> @ArcturusZhang is there anything in the arm automation that needs to be updated so that the "v" doesn't show up in the changelog entries?

Some script of build tool need to be updated. I wonder if this change is just applied to changelog as "v" is needed in version tag(https://golang.org/doc/modules/version-numbers)? @weshaggard

**Comment by chidozieononiwu** [comment]
> > @ArcturusZhang is there anything in the arm automation that needs to be updated so that the "v" doesn't show up in the changelog entries?
> 
> Some script of build tool need to be updated. I wonder if this change is just applied to changelog? As far as I'm concerned, "v" is a must in go package version (https://golang.org/doc/modules/version-numbers). @weshaggard @jhendrixMSFT

the change is applied only to CHANGELOG.md files. Release tags and version files will still contain the v

**Comment by seankane-msft** [comment]
@jhendrixMSFT can I update synapse to use the newer format (`go:build go1.16`) in this PR? Would this change break anything on the synapse side.

**Comment by weshaggard** [comment]
>Some script of build tool need to be updated.

@tadelesh what script will need to be updated based on this change to the changelog version format?

**Comment by jhendrixMSFT** [comment]
@seankane-msft updating the build constraints is fine.

File: eng/pipelines/templates/steps/analyze.yml [hunk 1]
```diff
@@ -27,7 +27,9 @@ steps:
   - pwsh: |
       $modDirs = ./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE)
       foreach ($md in $modDirs) {
-        Get-ChildItem $md/doc.go
+        if (-Not $md -Match "/arm") {
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:30]: Once my other change goes in we should probably switch this to using the type property from the metadata we read.

**Review by weshaggard — APPROVED** [review]
Looks Good. 

@tadelesh can you please make a follow-up PR with whatever changes are needed to update the automation for the changelogs. Thanks.

**Comment by tadelesh** [comment]
> Looks Good.
> 
> @tadelesh can you please make a follow-up PR with whatever changes are needed to update the automation for the changelogs. Thanks.

Just need some changes in mgmt release tool. https://github.com/Azure/azure-sdk-for-go/pull/15579

**Pull request merged**